### PR TITLE
[ALLUXIO-2556] Use current class loader when initializing AbstractFileSystem

### PR DESCRIPTION
**Pull request by apc999**
https://alluxio.atlassian.net/browse/ALLUXIO-2556

### CODE REVIEW

**Comment by apc999** [comment]
Please don't merge

File: core/client/src/main/java/alluxio/hadoop/AbstractFileSystem.java [hunk 1]
```diff
@@ -422,10 +422,10 @@ public Path getWorkingDirectory() {
   @SuppressFBWarnings("ST_WRITE_TO_STATIC_FROM_INSTANCE_METHOD")
   @Override
   public void initialize(URI uri, org.apache.hadoop.conf.Configuration conf) throws IOException {
-    // NOTE, we must switch the context classloader to the one provided by Hadoop configuration
-    // first before anything else. This ensures all Alluxio classes are loaded by the same
-    // classloader, given this class is already loaded by the Hadoop configuration classloader.
-    Thread.currentThread().setContextClassLoader(conf.getClassLoader());
+    // NOTE, switch the context classloader to the one load this class.
```
Comment (jsimsa) [inline core/client/src/main/java/alluxio/hadoop/AbstractFileSystem.java]: `NOTE: switch the context class loader to the one use for loading this class.

File: core/client/src/main/java/alluxio/hadoop/AbstractFileSystem.java [hunk 2]
```diff
@@ -422,10 +422,10 @@ public Path getWorkingDirectory() {
...
+    // This ensures all Alluxio classes are loaded by the same
+    // classloader, given this class is likely to be loaded by reflection from hadoop.
```
Comment (jsimsa) [inline core/client/src/main/java/alluxio/hadoop/AbstractFileSystem.java]: class loader

**Comment by ifcharming** [comment]
LGTM

File: core/client/src/main/java/alluxio/hadoop/AbstractFileSystem.java (hunk 1 above)
Comment (apc999) [inline core/client/src/main/java/alluxio/hadoop/AbstractFileSystem.java]: fixed

File: core/client/src/main/java/alluxio/hadoop/AbstractFileSystem.java (hunk 2 above)
Comment (apc999) [inline core/client/src/main/java/alluxio/hadoop/AbstractFileSystem.java]: fixed

**Comment by jsimsa** [comment]
doc changes LGTM

**Comment by aaudiber** [comment]
@apc999 do you think this is ready to merge?

**Comment by yupeng9** [comment]
@apc999 ping.

**Comment by apc999** [comment]
This issue has been solved by https://github.com/Alluxio/alluxio/pull/5302

**Pull request closed without merging**